src/main/webapp/map.js

```diff
@@ -1,43 +1,85 @@
-let map;
+var map;
+var infoWindow;
+var shopInfo;
+var userPos;
 
-/* Editable marker that displays when a user clicks in the map. */
-let editMarker;
-
-let markers = [[40.7122879, -74.0106218, "Starbucks"], [40.7105991, -74.0112465, "Irving Farm New York"]]
-
-/** Creates a map that allows users to add markers. */
+/** Creates a map that shows all coffee shops around the user. */
 function createMap() {
+  // Set default location at new york city
+  var newyork = new google.maps.LatLng(40.7128, -74.0060);
   map = new google.maps.Map(
     document.getElementById('map'),
-    {center: {lat: 40.7128, lng: -74.0060}, zoom: 12});
+    {center: newyork, zoom: 14});
+  
+  // Infowindow for to handle error in get user location
+  infoWindow = new google.maps.InfoWindow();
+  // Global infowindow for coffee shop
+  shopInfo = new google.maps.InfoWindow();
 
-  // When the user clicks in the map, show a marker with a text box the user can
-  // edit.
-  map.addListener('click', (event) => {
-    createMarkerForEdit(event.latLng.lat(), event.latLng.lng());
-  });
-  fetchMarkers();
+  // Try HTML5 geolocation to get user location.
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(
+      function(position) {
+        userPos = {
+          lat: position.coords.latitude,
+          lng: position.coords.longitude
+        };
+        map.setCenter(userPos);
+
+        // Request nearby coffee shop info
+        var request = {
+          location: userPos,
+          radius: '500',
```

Comment:
where are the units for this radius specified?

Reply:
Added a comment specifying the unit (metres)